plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -284,13 +284,26 @@ public T instantiate(Map<String,?> arguments) throws Exception {
             arguments = Collections.singletonMap(rp.getName(),arguments.get(ANONYMOUS_KEY));
         }
 
+        // Check for erroneous arguments
+        List<String> erroneous = new ArrayList<>();
+        for (String arg : arguments.keySet()) {
+            if (!parameters.containsKey(arg)) {
+                erroneous.add(arg);
+            }
+        }
+        if (erroneous.size() > 0) {
+            throw new IllegalArgumentException("Unknown parameter(s) found for class type '" +
```

Comment:
It feels risky to me to do this everywhere in case there are existing problems that we don't know about. Maybe only throw an exception if `Main.isUnitTest` is true, and log a warning otherwise? Could be controlled by system property like `DescribableModel.STRICT_PARAMETER_CHECKING` or something. I am curious what other reviewers think.

Reply:
I'll go ahead and issue a logger warning